### PR TITLE
[dy] Add upsert to MySQL, BQ, and Snowflake

### PR DESCRIPTION
**Pull request by dy46**
# Description

Add ability to use upsert in MySQL, BigQuery, and Snowflake IO classes. The implementation is based on the destinations in the mage_integrations module, but modified slightly to work with the IO classes.

# How Has This Been Tested?

- [x] Tested by upserting conflicting rows into BQ, MySQL, and Snowflake tables

# Checklist
- [x] The PR is tagged with proper labels (bug, enhancement, feature, documentation)
- [x] I have performed a self-review of my own code
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] If new documentation has been added, relative paths have been added to the appropriate section of `docs/mint.json`

cc:

### CODE REVIEW

File: mage_ai/io/base.py [hunk 1]
```diff
@@ -51,6 +51,7 @@ class ExportWritePolicy(str, Enum):
     APPEND = 'append'
     FAIL = 'fail'
     REPLACE = 'replace'
+    UPSERT = 'upsert'
```
Comment (wangxiaoyou1993) [inline mage_ai/io/base.py:54]: do we need another policy?
the merge command is applied in postgres and mssql when policy='append' and unique constraints are specified.

**Comment by wangxiaoyou1993** [comment]
Have you tested all three databases?

File: mage_ai/io/base.py (hunk 1 above)
Comment (dy46) [reply mage_ai/io/base.py:54]: ok i can do that for consistency

**Comment by dy46** [comment]
yes i tested all 3

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**